tagstudio/src/qt/main_window.py

```diff
@@ -37,6 +40,9 @@ class Ui_MainWindow(QMainWindow):
     def __init__(self, driver: "QtDriver", parent=None) -> None:
         super().__init__(parent)
         self.driver: "QtDriver" = driver
+        # temporarily putting driver to application property
+        (QApplication.instance() or self.parent()).setProperty("driver", driver)
+        theme.update_palette()  # update palette according to theme settings
```

Comment:
that's because this file was compiled from `home.ui` which is an outdated UI design file. changes have already been made to this file after compilation. i don't think we need to rely on the `home.ui` file anymore.
i'm not sure how to access the global settings atm. so this is just temporary.
@CyanVoxel, do you think we should still rely on `home.ui`?

Reply:
`home.ui` hasn't been used or touched since before the project went public - it's leftover from my testing with Qt Designer and I've been meaning to remove it from the project entirely in favor of just in-code widgets.

Actually, it seems like someone [did](https://github.com/TagStudioDev/TagStudio/blame/main/tagstudio/src/qt/ui/home.ui) try to use/update it, but I didn't pay it much mind since it's otherwise deprecated. So still, it should just be deleted.

Reply:
Isn't that the QMainWindow of the interface? Can it be removed easily? Opening the file in QtCreator shows it contains the search bar and related buttons and central thumbnail scroll area widget. 

The last change to the file was properly compiled from qtcreator, the updated compiler version comment and changes to the .ui file reflects that.

Reply:
I don't use QtCreator nor QtDesigner (after my initial exploration of it), so the `.ui` file(s) are of no use to the project as they aren't used to compile any interfaces that we use. [`main_window.py`](https://github.com/TagStudioDev/TagStudio/blame/main/tagstudio/sr) is the actual code that gets run, and has been the file I've edited manually since releasing the project. Outside of the one contributor who seems to have tried to go though `home.ui` which touched [`home_ui.py`](https://github.com/TagStudioDev/TagStudio/blame/main/tagstudio/src/qt/ui/home_ui.py), both `home.ui` and `home_ui.py` are not used by the project and serve no importance. The warning inside `main_window.py` should also be removed to reflect this.

While I knew about `home.ui` being left around, I didn't realize that it was confusingly coming with a `home_ui.py` file that we had copied and started to use as `main_window.py`. Stuff like this is why I wanted to get a UI refactor going, to ensure all the files are in relevant locations and there isn't any confusing trails of testing left behind from the early days.

Reply:
My b 🫡 Never really looked into editting the main window and always just assumed the files in the ui folder were used

Reply:
> Stuff like this is why I wanted to get a UI refactor going

refectoring this file on my abandoned list, which is why i mentioned changes to this as temporarily.